File: src/main/java/io/github/oleksivio/telegram/bot/core/controller/annotations/parser/unpacker/EditedChannelPostUnpackerParser.java
package io.github.oleksivio.telegram.bot.core.controller.annotations.parser.unpacker;

import org.springframework.stereotype.Component;
import io.github.oleksivio.telegram.bot.api.annotations.receiver.EditedChannelPostReceiver;
import io.github.oleksivio.telegram.bot.api.model.objects.std.Message;
import io.github.oleksivio.telegram.bot.core.controller.handler.unpack.UpdateUnpacker;

import java.util.Optional;

@Component
public class EditedChannelPostUnpackerParser extends UnpackerParser<EditedChannelPostReceiver, Message> {

    @Override
    public Class<EditedChannelPostReceiver> getAnnotationClass() {
        return EditedChannelPostReceiver.class;
    }

    @Override

    public UpdateUnpacker<Message> getUpdateUnpacker() {
        return new UpdateUnpacker<>(in -> Optional.ofNullable(in.getChannelPost()), Message.class);
    }

}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/core/controller/annotations/parser/unpacker/EditedChannelPostUnpackerParser.kt
package io.github.oleksivio.telegram.bot.core.controller.annotations.parser.unpacker

import io.github.oleksivio.telegram.bot.api.annotations.receiver.EditedChannelPostReceiver
import io.github.oleksivio.telegram.bot.api.model.objects.std.Message
import io.github.oleksivio.telegram.bot.core.controller.handler.unpack.UpdateUnpacker
import org.springframework.stereotype.Component

@Component
class EditedChannelPostUnpackerParser : UnpackerParser<EditedChannelPostReceiver, Message>() {
    override val updateUnpacker: UpdateUnpacker<Message>
        get() = UpdateUnpacker({ it.editedChannelPost }, Message::class)

    override val parserAnnotationClass = EditedChannelPostReceiver::class

}



